### PR TITLE
Add `--no-outer-gaps` to the `fullscreen` command

### LINKED ISSUE #373: Implement `fullscreen --no-outer-gaps`
When in (non-native) fullscreen mode, I typically want to disable gaps to maximize screen usage. It would be nice if gaps could be customized for fullscreen, just like they can with monitor patterns.

Related, it would also be nice to optionally treat workspaces with a single window as fullscreen, removing gaps.

### PR DESCRIPTION
**Pull request by vrumger**
Closes #373

### CODE REVIEW

File: Sources/AppBundle/model/MonitorEx.swift [hunk 1]
```diff
@@ -10,6 +10,15 @@ extension Monitor {
...
     }
 
+    var visibleRectWithoutOuterGaps: Rect {
+        return Rect(
+            topLeftX: visibleRect.topLeftCorner.x,
+            topLeftY: visibleRect.topLeftCorner.y,
+            width: visibleRect.width,
+            height: visibleRect.height
+        )
+    }
```
Comment (nikitabobko) [inline Sources/AppBundle/model/MonitorEx.swift:20]: Unnecessary property. You can simply use `visibleRect`

**Review by nikitabobko — CHANGES_REQUESTED** [review]

**Comment by vrumger** [comment]
thanks for the suggestions, I fixed all of them

File: Sources/Common/cmdArgs/FullscreenCmdArgs.swift [hunk 2]
```diff
@@ -5,17 +5,22 @@ public struct FullscreenCmdArgs: CmdArgs, RawCmdArgs {
         kind: .fullscreen,
         allowInConfig: true,
         help: """
-            USAGE: fullscreen [-h|--help] [on|off]
+            USAGE: fullscreen [-h|--help] [--no-outer-gaps] [on|off]
```
[the remark this answers is not in the document]
Comment (nikitabobko) [reply Sources/Common/cmdArgs/FullscreenCmdArgs.swift]: 1. You forgot to update synopsis in `.adoc`
2. You forgot to update the arguments parser to disallow invalid combination of `--no-outer-gaps` and `off` arguments. Please take a look at how invalid combinations of args are reported in other commands and do the same (e.g. `FocusCmdArgs`)

File: grammar/commands-bnf-grammar.txt [hunk 3]
```diff
@@ -25,7 +25,8 @@ aerospace -h;
     | focus-monitor [--wrap-around] (next|prev) [--wrap-around]
     | focus-monitor <monitor_pattern>...
 
-    | fullscreen [on|off]
+    | fullscreen [--no-outer-gaps] [on]
```
Comment (nikitabobko) [inline grammar/commands-bnf-grammar.txt:28]: Shell completion should allow `on` and `--no-outer-gaps` in different order

```suggestion
    | fullscreen [--no-outer-gaps] [on]
    | fullscreen [on] [--no-outer-gaps]
```

**Comment by nikitabobko** [comment]
Thanks!

Merged with the following patch:
```diff
diff --git a/Sources/AppBundle/command/FullscreenCommand.swift b/Sources/AppBundle/command/FullscreenCommand.swift
index 6a758b9b..71b70bab 100644
--- a/Sources/AppBundle/command/FullscreenCommand.swift
+++ b/Sources/AppBundle/command/FullscreenCommand.swift
@@ -14,7 +14,7 @@ struct FullscreenCommand: Command {
             case .off: false
             case .toggle: !window.isFullscreen
         }
-        if newState == window.isFullscreen && args.noOuterGaps == window.noOuterGapsInFullscreen {
+        if newState == window.isFullscreen {
             return state.failCmd(msg: newState ? "Already fullscreen" : "Already not fullscreen")
         }
         window.isFullscreen = newState
diff --git a/Sources/Common/cmdArgs/FullscreenCmdArgs.swift b/Sources/Common/cmdArgs/FullscreenCmdArgs.swift
index f865691a..119fee05 100644
--- a/Sources/Common/cmdArgs/FullscreenCmdArgs.swift
+++ b/Sources/Common/cmdArgs/FullscreenCmdArgs.swift
@@ -28,5 +28,5 @@ public struct FullscreenCmdArgs: CmdArgs, RawCmdArgs {
 
 public func parseFullscreenCmdArgs(_ args: [String]) -> ParsedCmd<FullscreenCmdArgs> {
     return parseRawCmdArgs(FullscreenCmdArgs(rawArgs: args), args)
-        .filter("--no-outer-gaps is incompatible with node off") { $0.toggle != .off || $0.noOuterGaps != true }
+        .filter("--no-outer-gaps is incompatible with 'off' argument") { $0.toggle != .off || $0.noOuterGaps != true }
 }
diff --git a/docs/aerospace-fullscreen.adoc b/docs/aerospace-fullscreen.adoc
index 33beda8d..54857b14 100644
--- a/docs/aerospace-fullscreen.adoc
+++ b/docs/aerospace-fullscreen.adoc
@@ -7,9 +7,12 @@ include::util/man-attributes.adoc[]
 
 // =========================================================== Synopsis
 == Synopsis
+
+[verse]
 // tag::synopsis[]
 aerospace fullscreen [-h|--help] [--no-outer-gaps] [on]
 aerospace fullscreen [-h|--help] off
+
 // end::synopsis[]
 
 // =========================================================== Description

```

**Pull request closed without merging**